### PR TITLE
feat: add sanity precheck for contract call

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Add sanity pre-checks to verify ContractCall transaction semantics.  

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Neeharika-Sompalli — DISMISSED** [review]
Makes sense to me. Thanks @lukelee-sl ! Will defer to @tinker-michaelj's review

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java [hunk 1]
```diff
@@ -1569,6 +1572,43 @@ HapiSpec invalidContract() {
                 .then(contractCallWithFunctionAbi("invalid", function).hasKnownStatus(INVALID_CONTRACT_ID));
     }
 
+    // This test for modularization
+    HapiSpec invalidContractDoesNotExist() {
+        final AtomicReference<AccountID> accountID = new AtomicReference<>();
+        final var function = getABIFor(FUNCTION, "getIndirect", CREATE_TRIVIAL);
+
+        return defaultHapiSpec("invalidContractDoesNotExist")
+                .given(cryptoCreate("invalid")
```
Comment (tinker-michaelj) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java:1581]: Could remove the `cryptoCreate()` here for clarity.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java [hunk 2]
```diff
@@ -1569,6 +1572,43 @@ HapiSpec invalidContract() {
...
+                        .exposingCreatedIdTo(accountID::set))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        contractCallWithFunctionAbi("0.0." + accountID.get().getAccountNum(), function)
+                                .hasPrecheck(INVALID_CONTRACT_ID)
+                                .via("contractIsAccount"))))
+                .then(getTxnRecord("contractIsAccount").hasAnswerOnlyPrecheck(RECORD_NOT_FOUND));
+    }
+
+    // This test disabled for modularization service
```
Comment (tinker-michaelj) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java:1611]: Unintentional comment?

**Review by tinker-michaelj — DISMISSED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java (hunk 2 above)
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java:1629]: removed.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java]: yes, makes sense.  removed.

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**